src/ray/gcs/redis_gcs_client.h

```diff
@@ -19,7 +19,11 @@ class RedisContext;
 
 class RAY_EXPORT RedisGcsClient : public GcsClientInterface {
   friend class ActorStateAccessor;
+  friend class JobStateAccessor;
   friend class SubscriptionExecutorTest;
+  friend void TestLogSubscribeAll(const JobID &, std::shared_ptr<gcs::RedisGcsClient>);
+  friend void TestLogSubscribeId(const JobID &, std::shared_ptr<gcs::RedisGcsClient>);
+  friend void TestLogSubscribeCancel(const JobID &, std::shared_ptr<gcs::RedisGcsClient>);
```

Comment:
It's not a good practice to declare a `Test` method as a `friend` in my sense, but I'm not sure.

@zhijunfu @pcmoritz  How do you think of this?

Reply:
This looks a bit strange to me too... Would it work if we instead put the test class as friend?

Reply:
Actually those test didn't belong to any test class.

Reply:
Maybe I can move those tests to a new test Class.

Reply:
Already add test helper class `LogSubscribeTestHelper` as friend class of `RedisGcsClient`.
Please let me known if you guys has any suggestion.